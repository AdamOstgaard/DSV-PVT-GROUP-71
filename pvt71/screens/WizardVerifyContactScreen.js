import React from "react";
import AppSingleButton from "../components/AppSingleButton";
import { Animated, Dimensions, Keyboard, UIManager ,StyleSheet, Text, View, TextInput, AsyncStorage } from "react-native";
import { getStatusBarHeight } from "react-native-status-bar-height";

const { State: TextInputState } = TextInput;
export default class WizardVerifyContactScreen extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      name: '', 
      number: '',
      shift: new Animated.Value(0),
    };
  }
  componentWillMount() {
    this.keyboardDidShowSub = Keyboard.addListener('keyboardDidShow', this.handleKeyboardDidShow);
    this.keyboardDidHideSub = Keyboard.addListener('keyboardDidHide', this.handleKeyboardDidHide);
  }

  componentWillUnmount() {
    this.keyboardDidShowSub.remove();
    this.keyboardDidHideSub.remove();
  }
  render() {
    const { shift } = this.state;
    return (
      <Animated.View style={[styles.container, { transform: [{translateY: shift}] }]}>
        <View style={styles.headerContainer}>
          <Text style={styles.headerText}>Lägg till kontaktperson</Text>
          <Text style={styles.infoText}>
            Du kan alltid ändra och lägga till kontakpersoner senare under inställningar 
          </Text>
        </View>
        <View style = {styles.nameContainer}>
          <Text style = {styles.infoText}>Namn: </Text>
          <TextInput style={styles.textInput}
          placeholder="Ange namn här!"
          onChangeText={(name) => this.setState({name})}
          value={this.state.name}
          />
        </View>
        <View style = {styles.numberContainer}>
          <Text style={styles.infoText}>Telefonnummer: </Text>
          <TextInput style={styles.textInput}
          placeholder="ange nummer här!"
          onChangeText={(number) => this.setState({number})}
          value={this.state.number}
        />
        </View>
        <View style={styles.bottom}>
          <AppSingleButton
            title="Nästa"
            onPress={() => {
              this.saveSettings();
              this.props.navigation.navigate("WizardSettingsScreen");
            }}
          />
        </View>
      </Animated.View>
    );
  }
  handleKeyboardDidShow = (event) => {
    const { height: windowHeight } = Dimensions.get('window');
    const keyboardHeight = event.endCoordinates.height;
    const currentlyFocusedField = TextInputState.currentlyFocusedField();
    UIManager.measure(currentlyFocusedField, (originX, originY, width, height, pageX, pageY) => {
      const fieldHeight = height;
      const fieldTop = pageY;
      const gap = (windowHeight - keyboardHeight) - (fieldTop + fieldHeight);
      if (gap >= 0) {
        return;
      }
      Animated.timing(
        this.state.shift,
        {
          toValue: gap,
          duration: 1000,
          useNativeDriver: true,
        }
      ).start();
    });
  }
  handleKeyboardDidHide = () => {
    Animated.timing(
      this.state.shift,
      {
        toValue: 0,
        duration: 1000,
        useNativeDriver: true,
      }
    ).start();
  }


  async saveSettings() {
    let contactPerson = {
      name: this.state.name,
      number: this.state.number
    }
    try {
      await AsyncStorage.setItem("contact", JSON.stringify(contactPerson));
    } catch (error) {

    }
  }
}

const styles = StyleSheet.create({
  container: {
    marginTop: getStatusBarHeight(),
    flex: 1,
    backgroundColor: "#fff"
  },
  headerContainer: {
    alignItems:'flex-start',
    marginBottom: 10
  },
  headerText: {
    fontSize: 30,
    justifyContent: "flex-start",
    left: 10,
  },

  nameContainer:{
    marginTop: 20,
    marginBottom: 20,
    alignItems:'flex-start',
    
  },
  numberContainer:{
    alignItems:'flex-start',
  
    
  },
  infoText: {
    fontSize: 20,
    paddingLeft: 10
  },
  textInput: {
    height: 60,
    paddingLeft: 10,
    marginLeft: 10,
    marginRight: 10,
    width: "90%",
    borderColor: 'gray',
    borderWidth: 1
  },
  bottom: {
    flex: 1,
    justifyContent: "flex-end"
  }
});