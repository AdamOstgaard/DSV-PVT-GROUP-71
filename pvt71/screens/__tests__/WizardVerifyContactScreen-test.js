import "react-native";
import React from "react";
import  WizardVerifyContactScreen  from "../WizardVerifyContactScreen";
import renderer from "react-test-renderer";

describe("Wizard Verify Contact Screen tests", () => {
    describe("Snapshot test", () => {
        it("renders correctly", () => {
        const tree = renderer.create(<WizardVerifyContactScreen />).toJSON();
        expect(tree).toMatchSnapshot();
        });
    });
});