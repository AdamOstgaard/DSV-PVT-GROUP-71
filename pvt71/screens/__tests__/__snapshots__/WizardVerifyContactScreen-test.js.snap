// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wizard Verify Contact Screen tests Snapshot test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "marginTop": 20,
      "transform": Array [
        Object {
          "translateY": 0,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "marginBottom": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 30,
          "justifyContent": "flex-start",
          "left": 10,
        }
      }
    >
      Lägg till kontaktperson
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
          "paddingLeft": 10,
        }
      }
    >
      Du kan alltid ändra och lägga till kontakpersoner senare under inställningar
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  >
<<<<<<< HEAD
     
    Nödkontakt efternamn:
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 20,
        "marginLeft": 10,
        "marginRight": 10,
        "paddingLeft": 10,
=======
    <Text
      style={
        Object {
          "fontSize": 20,
          "paddingLeft": 10,
        }
>>>>>>> 797930c7bd25fcdd9d6c4091931013a9b870b922
      }
    >
      Namn: 
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Ange namn här!"
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 60,
          "marginLeft": 10,
          "marginRight": 10,
          "paddingLeft": 10,
          "width": "90%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "paddingLeft": 10,
        }
      }
    >
      Telefonnummer: 
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="ange nummer här!"
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 60,
          "marginLeft": 10,
          "marginRight": 10,
          "paddingLeft": 10,
          "width": "90%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0d6f73",
          "height": 60,
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 30,
            "paddingRight": 10,
            "paddingTop": 8,
            "textAlign": "right",
          }
        }
      >
        Nästa
      </Text>
    </View>
  </View>
</View>
`;
